app/models/provider.rb

```diff
@@ -37,4 +37,8 @@ def accredited_courses
   def application_forms
     ApplicationForm.where(id: application_choices.select(:application_form_id))
   end
+
+  def onboarded?
+    ProviderAgreement.find_by(provider: self).present?
```

Comment:
```suggestion
    provider_agreements.any?
```

results in a more efficient query:

```
[1] pry(main)> ProviderAgreement.find_by(provider: Provider.last).present?
  Provider Load (1.8ms)  SELECT "providers".* FROM "providers" ORDER BY "providers"."id" DESC LIMIT $1  [["LIMIT", 1]]
  ProviderAgreement Load (0.4ms)  SELECT "provider_agreements".* FROM "provider_agreements" WHERE "provider_agreements"."provider_id" = $1 LIMIT $2  [["provider_id", 5], ["LIMIT", 1]]
=> false
[2] pry(main)> Provider.last.provider_agreements.any?
  Provider Load (0.4ms)  SELECT "providers".* FROM "providers" ORDER BY "providers"."id" DESC LIMIT $1  [["LIMIT", 1]]
  ProviderAgreement Exists? (0.4ms)  SELECT 1 AS one FROM "provider_agreements" WHERE "provider_agreements"."provider_id" = $1 LIMIT $2  [["provider_id", 5], ["LIMIT", 1]]
=> false
```

Reply:
oops - yes!